### PR TITLE
[SMTChecker] Organize SMTChecker.h/cpp

### PR DESCRIPTION
**Pull request by leonardoalt**
Just group functions together for better internal organization. It was a mess.

This PR does not remove or add any code, just reorganizes.

### CODE REVIEW

**Comment by chriseth** [comment]
This is non-reviewable...

File: libsolidity/formal/SMTChecker.cpp [hunk 1]
```diff
@@ -75,6 +75,8 @@ void SMTChecker::analyze(SourceUnit const& _source, shared_ptr<Scanner> const& _
 	m_errorReporter.clear();
 }
 
+/** Visit functions. */
```
Comment (chriseth) [inline libsolidity/formal/SMTChecker.cpp]: doxygen has some grouping functionality: http://www.doxygen.nl/manual/grouping.html
Also please use `///` unless you are documenting a class.

File: libsolidity/formal/SMTChecker.h [hunk 2]
```diff
@@ -269,17 +300,16 @@ class SMTChecker: private ASTConstVisitor
...
 	std::vector<FunctionDefinition const*> m_functionPath;
 	/// Stores the current call/invocation path.
 	std::vector<ASTNode const*> m_callStack;
-	/// Returns true if the current function was not visited by
-	/// a function call.
-	bool isRootFunction();
-	/// Returns true if _funDef was already visited.
-	bool visitedFunction(FunctionDefinition const* _funDef);
 
+	/** Overflow. */
```
Comment (chriseth) [inline libsolidity/formal/SMTChecker.h]: ///

File: libsolidity/formal/SMTChecker.cpp (hunk 1 above)
Comment (leonardoalt) [reply libsolidity/formal/SMTChecker.cpp]: Changed to the recommended style

**Comment by leonardoalt** [comment]
@chriseth updated

**Comment by leonardoalt** [comment]
Yea I know... the diff is quite bad

**Comment by axic** [comment]
> Yea I know... the diff is quite bad

I suggest you do it in smaller steps, it may take 5-10 PRs to get there.

**Comment by leonardoalt** [comment]
Yea, not sure it's worth it in that case, it doesn't really add anything.

**Pull request closed without merging**